packages/next/src/server/lib/incremental-cache/index.ts

```diff
@@ -468,12 +469,15 @@ export class IncrementalCache implements IncrementalCacheType {
         )
       }
 
+      const workStore = workAsyncStorage.getStore()
       const combinedTags = [...(ctx.tags || []), ...(ctx.softTags || [])]
       // if a tag was revalidated we don't return stale data
       if (
-        combinedTags.some((tag) => {
-          return this.revalidatedTags?.includes(tag)
-        })
+        combinedTags.some(
+          (tag) =>
+            this.revalidatedTags?.includes(tag) ||
+            workStore?.pendingRevalidatedTags?.includes(tag)
+        )
```

Comment:
incremental cache doesn't use react.cache does it? So we already had a situation where the moment we have some revalidatedTags from outside the entire render would never cache that entry? and now the moment you do an in-action revalidate all associated cache entries similarly will no longer be readable from the cache?

It feels like we need some notion of time to be modeled here. Or we need to rely on react.cache to add a layer of per-request caching.

but actually react.cache is a problem here I think because we'd need to invalidate it after you call any `revalidate*()` method since it doesn't know if it is holding onto a value that is no longer legit.

Hmmm this is tricky